src/main/java/in/ac/amu/zhcet/controller/DepartmentController.java

```diff
@@ -35,19 +44,21 @@ public String createCourse(@ModelAttribute Course course) {
         // TODO: Add form handling
         departmentAdminService.registerCourse(course);
 
-        return "redirect:department";
+        return "redirect:/department";
     }
 
     @PostMapping("/department/float_course")
-    public String floatCourse(@ModelAttribute Course course) {
+    public String floatCourse(@RequestParam String course, @RequestParam List<String> faculty) {
         // TODO: Add form handling
+        Course course1 = courseRepository.findByCode(course);
+
         try {
-            departmentAdminService.floatCourse(course);
+            departmentAdminService.floatCourse(course1, faculty);
```

Comment:
Move `courseRepository.findByCode(course);` in departmentAdminService

All repository work should be done there